packages/client/src/Call.ts

```diff
@@ -483,74 +502,74 @@ export class Call {
     reject = false,
     reason = 'user is leaving the call',
   }: CallLeaveOptions = {}) => {
-    const callingState = this.state.callingState;
-    if (callingState === CallingState.LEFT) {
-      throw new Error('Cannot leave call that has already been left.');
-    }
-
-    if (callingState === CallingState.JOINING) {
-      await this.assertCallJoined();
-    }
-
-    this.isLeaving = true;
+    await withoutConcurrency(this.joinLeaveConcurrencyTag, async () => {
+      const callingState = this.state.callingState;
+      if (callingState === CallingState.LEFT) {
+        throw new Error('Cannot leave call that has already been left.');
+      }
 
-    if (this.ringing) {
-      // I'm the one who started the call, so I should cancel it.
-      const hasOtherParticipants = this.state.remoteParticipants.length > 0;
-      if (
-        this.isCreatedByMe &&
-        !hasOtherParticipants &&
-        callingState === CallingState.RINGING
-      ) {
-        // Signals other users that I have cancelled my call to them
-        // before they accepted it.
-        await this.reject();
-      } else if (reject && callingState === CallingState.RINGING) {
-        // Signals other users that I have rejected the incoming call.
-        await this.reject();
+      if (callingState === CallingState.JOINING) {
+        await this.assertCallJoined();
       }
-    }
 
-    this.statsReporter?.stop();
-    this.statsReporter = undefined;
+      this.isLeaving = true;
 
-    this.sfuStatsReporter?.stop();
-    this.sfuStatsReporter = undefined;
+      if (this.ringing) {
+        // I'm the one who started the call, so I should cancel it.
+        const hasOtherParticipants = this.state.remoteParticipants.length > 0;
+        if (
+          this.isCreatedByMe &&
+          !hasOtherParticipants &&
+          callingState === CallingState.RINGING
+        ) {
+          // Signals other users that I have cancelled my call to them
+          // before they accepted it.
+          await this.reject();
+        } else if (reject && callingState === CallingState.RINGING) {
+          // Signals other users that I have rejected the incoming call.
+          await this.reject();
+        }
+      }
 
-    this.subscriber?.close();
-    this.subscriber = undefined;
+      this.statsReporter?.stop();
+      this.statsReporter = undefined;
 
-    this.publisher?.close();
-    this.publisher = undefined;
+      this.sfuStatsReporter?.stop();
+      this.sfuStatsReporter = undefined;
 
-    this.sfuClient?.close(StreamSfuClient.NORMAL_CLOSURE, reason);
-    this.sfuClient = undefined;
+      this.subscriber?.close();
+      this.subscriber = undefined;
 
-    this.dispatcher.offAll();
```

Comment:
🍓 Not removing listeners from dispatcher.

That seems to be safe, since every object keeping a reference to a dispatcher (except for the call itself) is nullified on leave. So the call instance will be GC-ed properly even without `offAll`.